src/main/groovy/org/mockito/release/internal/gradle/E2ETestingPlugin.java

```diff
@@ -0,0 +1,100 @@
+            this.project = project;
+        }
+
+        void create(String gitHubRepoUrl) {
+            String repoName = extractRepoName(gitHubRepoUrl);
+            CloneGitRepositoryTask clone = createCloneProjectFromGitHub(gitHubRepoUrl, repoName);
+            CloneGitRepositoryTask workDirCloneTask = createCloneProjectToWorkDirTask(repoName, clone);
+            createRunTestReleaseTask(repoName, workDirCloneTask);
+        }
+
+        private CloneGitRepositoryTask createCloneProjectFromGitHub(String gitHubRepoUrl, String repoName) {
+            // TODO add depth clone configuration for shallow clone
+            CloneGitRepositoryTask clone = project.getTasks().create(
+                    "cloneProjectFromGitHub" + capitalize(repoName),
+                    CloneGitRepositoryTask.class);
+            clone.setRepository(gitHubRepoUrl);
+            clone.setTargetDir(new File(project.getBuildDir(), repoName + "-pristine"));
+            // For now for easier testing
+            clone.dependsOn("clean");
+            return clone;
+        }
+
+        private CloneGitRepositoryTask createCloneProjectToWorkDirTask(String repoName, CloneGitRepositoryTask clone) {
+            // Clone from *-pristine to *-work. Copy task will not work because of ignoring git specific files:
+            // https://discuss.gradle.org/t/copy-git-specific-files/11970
+            // Furthermore we can verify push to pristine origin
+            File workDir = new File(project.getBuildDir(), repoName + "-work");
+            CloneGitRepositoryTask copy = project.getTasks().create(
+                    "cloneProjectToWorkDir" + capitalize(repoName),
+                    CloneGitRepositoryTask.class);
+            copy.dependsOn(clone);
+            copy.setRepository(clone.getTargetDir().getAbsolutePath());
+            copy.setTargetDir(workDir);
+            return copy;
+        }
+
+        private void createRunTestReleaseTask(String repoName, CloneGitRepositoryTask copy) {
+            RunTestReleaseTask run = project.getTasks().create(
+                    "runTestRelease" + capitalize(repoName),
+                    RunTestReleaseTask.class);
+            run.dependsOn(copy);
+            run.setWorkDir(copy.getTargetDir());
+            run.setRepoName(repoName);
+
+            // Using Gradle's composite builds ("--include-build") so that we're picking up current version of tools
+            run.setCommand(asList("./gradlew", "publishToMavenLocal", "testRelease",
+                    "-x", "gitPush", "-x", "bintrayUpload",
+                    "--include-build", project.getRootDir().getAbsolutePath(), "-s"));
+
+            // Build log in separate file instead of including it in the console of the parent build
+            // Otherwise the output will be really messy
+            run.setBuildOutputFile(new File(project.getBuildDir(), repoName + "-build.log"));
+        }
+
+        private String extractRepoName(String gitHubRepo) {
```

Comment:
This is smart - makes the plugin figure out the details and user can just provide the url to the test.